src/fusion_power_plant_tests.cc

```diff
@@ -37,18 +64,442 @@ TEST_F(FusionPowerPlantTest, Print) {
   // Test FusionPowerPlant specific aspects of the print method here
 }
 
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-TEST_F(FusionPowerPlantTest, Tick) {
-  ASSERT_NO_THROW(facility->Tick());
-  // Test FusionPowerPlant specific behaviors of the Tick function here
-}
-
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 TEST_F(FusionPowerPlantTest, Tock) {
   EXPECT_NO_THROW(facility->Tock());
   // Test FusionPowerPlant specific behaviors of the Tock function here
 }
 
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+TEST_F(FusionPowerPlantTest, BlanketCycle) {
+  // Test that the agent correctly removes and replaces a portion of the
+  // blanket every blanket turnover period. The default period is 1 timestep
+  // so it is left undefined here.
+
+  std::string config =
+      " <fusion_power>300</fusion_power>"
+      "  <TBR>1.00</TBR>"
+      " <reserve_inventory>6.0</reserve_inventory>"
+      " <sequestered_equilibrium>2.121</sequestered_equilibrium>"
+      " <fuel_incommod>Tritium</fuel_incommod>"
+      " <blanket_incommod>Enriched_Lithium</blanket_incommod>"
+      " <blanket_outcommod>Depleted_Lithium</blanket_outcommod>"
+      " <blanket_inrecipe>enriched_lithium</blanket_inrecipe>"
+      " <blanket_size>1000</blanket_size>"
+      " <blanket_turnover_fraction>0.03</blanket_turnover_fraction>"
+      " <he3_outcommod>Helium_3</he3_outcommod>";
+
+  int simdur = 4;
+  cyclus::MockSim sim(cyclus::AgentSpec(":tricycle:FusionPowerPlant"), config, simdur);
+
+  sim.AddRecipe("tritium", tritium());
+  sim.AddRecipe("enriched_lithium", enriched_lithium());
+
+  sim.AddSource("Tritium").recipe("tritium").Finalize();
+  sim.AddSource("Enriched_Lithium").recipe("enriched_lithium").Finalize();
```

Comment:
There may be some value in also bundling all of this into a function since most of it happens the same with each test.  At least the following:
* setup the MockSim
* add two recipes
* add one source for enriched lithium

Reply:
I went in and did this for all the "standard" sims, but there were two or three that had non-standard implementations so I just left those alone instead of trying to be clever.